public/js/countyGraph.js

```diff
@@ -0,0 +1,190 @@
+const graph = document.getElementById('graph').getContext('2d');
+Chart.defaults.global.defaultFontSize = 10;
+
+const countyGraphChange = async (county) => {
+  const kenyaCounties = [
+    'Nairobi',
+    'Kwale',
+    'Mombasa',
+    'Kilifi',
+    'Tana River',
+    'Lamu',
+    'Taita Taveta',
+    'Garissa',
+    'Wajir',
+    'Mandera',
+    'Marsabit',
+    'Isiolo',
+    'Tharaka Nithi',
+    'Meru',
+    'Embu',
+    'Kitui',
+    'Machakos',
+    'Makueni',
+    'Nyanadarua',
+    'Nyeri',
+    'Kirinyaga',
+    'Muranga',
+    'Kiambu',
+    'Turkana',
+    'West Pokot',
+    'Samburu',
+    'Trans Nzoia',
+    'Uasin Gishu',
+    'Elgeyo Marakwet',
+    'Nandi',
+    'Baringo',
+    'Laikipia',
+    'Nakuru',
+    'Marok',
+    'Kajiado',
+    'Kericho',
+    'Bomet',
+    'Kakamega',
+    'Vihiga',
+    'Bungoma',
+    'Busia',
+    'Siaya',
+    'Kisumu',
+    'Homa Bay',
+    'Migori',
+    'Kisii',
+    'Nyamira',
+  ];
+  const countyData = await getCounties();
+
+  let countiesMap = {};
+
+  // Map total nodes per county. This will be used to populate the chart when counties are available
+  results.results.filter((result) => {
+    const cityCounty = getCityCounty(result.location.city, countyData);
+
+    // map of nodes per county
+    if (cityCounty) {
+      if (!countiesMap[cityCounty]) {
+        countiesMap[cityCounty] = '1';
+      } else {
+        countiesMap[cityCounty] = Number(countiesMap[cityCounty]) + 1;
+      }
+    }
+
+    return cityCounty === county;
+  });
+
+  const newArr = new Array(47 - Object.keys(countiesMap).length).fill(0); // populate the rest of the counties with 0 nodes
+
+  // Array of counties
+  const countiesArr = [
+    ...new Set([...Object.keys(countiesMap), ...kenyaCounties]),
+  ];
+  // Array of nodes in counties
+  const countiesDataArr = [...Object.values(countiesMap), ...newArr];
```

Comment:
Not sure about this computation here; It looks to me like you're counting cities in a county and not nodes (that should come from `results`) in a city in a county.

Reply:
I calculated the nodes per county (city is what I use to get the county it belongs to) from results [here](https://github.com/CodeForAfrica/sensors.AFRICA-Dashboard-KE-HAP/pull/10/files#diff-6cbf18c3bcff269253eee101b966c4d44588e940e870272dc883b2b98bdcd0a6R59) and put them in the object `countiesMap`